<!-- -*- mode: XML -*- -->
<launch>    

	<arg name="scen" default="scenario_1"/> 

	<arg name="scenarios_folder_url" value="$(find SRVSS)/work_space/$(arg scen)"/>
	<arg name="world_models_folder_url" value="$(find SRVSS)/world_components_models"/>
	<arg name="world_name" value="$(arg scenarios_folder_url)/scenarioEnv.world"/>
	<arg name="pause" value="-u"/>
	<arg name="record" value="-r"/>
	<arg name="record_file_path" value="-O $(arg scenarios_folder_url)/scen_rec.bag"/>
        <arg name="SFV_file_url" value="$(arg scenarios_folder_url)/scen.SFV"/>

<!-- setting gazebo path for scenario platform and sensors models -->
	<env name="GAZEBO_MODEL_PATH" value="$(arg world_models_folder_url):$(arg scenarios_folder_url)/scenarioSystemModels"/>

<!-- synchronizing ros to gazebo -->
	<param name="/use_sim_time" value="true" />

<!-- start gazebo server-->
	<node name="gazebo" pkg="gazebo_ros" type="gzserver" respawn="false" output="screen" args="$(arg world_name) "/>
	

<!-- start gazebo client -->
	<!--node name="gazebo_gui" pkg="gazebo_ros" type="gzclient" respawn="false" output="screen" /-->

<!-- Robil2_bobtank -->
<!---->
  	 <include file="$(find bobcat_model)/launch/bobcat_arm_controler.launch" />
	<!--param name="robot_description" command="cat $(find bobtank)/urdf/BOBCAT_sdf_model_new.URDF" / -->
	 
	<!-- include file="$(find bobtank)/launch/bobcat_gazebo_control_sdf_model _with_tracks.launch" / -->   <!-- the blank is needed (I dont know why) -->
	<!--node name="rate2effort_for_tracks" pkg="rate2effort" type="rate2effort_for_tracks" cwd="node" output="screen" /-->
	<node pkg="SRVSS" type="timed_roslaunch.sh" args="30 SRVSS runScenario_robil2_bobtank_server_sim2.launch scen:=$(arg scen)" name="timed_roslaunch" output="screen">   </node>
</launch>





